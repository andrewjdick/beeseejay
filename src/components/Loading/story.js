import React from "react";
import { storiesOf } from "@storybook/react";
import { boolean } from "@storybook/addon-knobs";
import { Loading } from "./index";

storiesOf("Components/Loading", module).add("default", () => (
  <Loading isLoading={boolean("is loading?", false)} />
));
