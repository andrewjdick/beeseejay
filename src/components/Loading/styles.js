import styled from "styled-components";

export const Container = styled.div`
  width: 40px;
  height: 40px;
`;

export const Spinner = styled.img`
  width: 100%;
  height: auto;
  animation: spinning 1.4s infinite linear;

  @keyframes spinning {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
