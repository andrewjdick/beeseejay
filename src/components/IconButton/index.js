import React from "react";
import { node } from "prop-types";
import { Button } from "./styles";

export const IconButton = ({ children, ...props }) => (
  <Button {...props}>{children}</Button>
);

IconButton.propTypes = {
  children: node.isRequired()
};
