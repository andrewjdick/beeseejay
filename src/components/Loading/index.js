import React from "react";
import brandLogo from "assets/images/logo.png";
import { Container, Spinner } from "./styles";

export const Loading = props => (
  <Container {...props}>
    <Spinner src={brandLogo} alt="Loading..." />
  </Container>
);
