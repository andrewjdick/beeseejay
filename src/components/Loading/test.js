import React from "react";
import renderer from "react-test-renderer";
import { Loading } from "components/Loading";

it("renders correctly", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders loading correctly", () => {
  const tree = renderer.create(<Loading isLoading={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
