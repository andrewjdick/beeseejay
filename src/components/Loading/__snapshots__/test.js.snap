// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa kcLQln"
>
  <img
    alt="BCG Digital Ventures logo"
    className="sc-bwzfXH hruRYZ"
    src="logo.png"
  />
</div>
`;

exports[`renders loading correctly 1`] = `
<div
  className="sc-bdVaJa kcLQln"
>
  <img
    alt="BCG Digital Ventures logo"
    className="sc-bwzfXH jZOxdD"
    src="logo.png"
  />
</div>
`;
